tests/test_signer.py

```diff
@@ -108,14 +116,14 @@ async def test_execute():
     assert execution_info.result == (3,)
 
 
+@pytest.mark.parametrize("network", NETWORKS)
 @pytest.mark.asyncio
-async def test_chain_id():
-    mainnet = Signer(PRIVATE_KEY, "mainnet")
-    assert mainnet.chain_id == StarknetChainId.MAINNET.value
-
-    testnet = Signer(PRIVATE_KEY, "testnet")
-    assert testnet.chain_id == StarknetChainId.TESTNET.value
+async def test_chain_id(network):
+    # check all networks
```

Comment:
i think it's confusing to say "all networks" and then check only one, i think the "all" part is implied in the fixture rather than here